specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,254 @@
+
+### API
+
+*Component name:*
+- `fast-anchored-region`
+
+*Attributes:*
+- anchorId - The html id of the HTMLElement used as the anchor around which the positioning region is placed.  This must be set for the component's positioning logic to be active.
+
+- viewportId - The html id of the HTMLElement used as the viewport the component uses to determine available layout space around the anchor element.  If unset the immediate parent element of the positioner is used.
+
+- horizontalPositioningMode - Can be 'uncontrolled', 'locktodefault', 'dynamic' or 'onetime'.  Default is 'uncontrolled'.
+- horizontalDefaultPosition - Can be 'left', 'right' or 'unset'.  Default is 'unset'
+- horizontalInset - Boolean that indicates whether the region should overlap the anchor on the horizontal axis. Default is false which places the region adjacent to the anchor element.
+- horizontalThreshold - Numeric value that defines how small in pixels the region must be to the edge of the viewport to switch to the opposite side of the anchor. The component favors the default position until this value is crossed.  When there is not enough space on either side or the value is unset the side with the most space is chosen.
+- horizontalScalingEnabled - The region is sized from code to match available space, in other scenarios the region gets sized via content size.
+
+- verticalPositioningMode - Can be 'uncontrolled', 'locktodefault', 'dynamic' or 'onetime'.  Default is 'uncontrolled'.
+- verticalDefaultPosition - Can be 'top', 'bottom' or 'unset'. Default is unset.
+- verticalInset - Boolean that indicates whether the region should overlap the anchor on the vertical axis. Default is false which places the region adjacent to the anchor element.
+- verticalThreshold - Numeric value that defines how small the region must be to the edge of the viewport to switch to the opposite side of the anchor. The component favors the default position until this value is crossed.  When there is not enough space on either side or the value is unset the side with the most space is chosen.
+- verticalScalingEnabled - The region is sized from code to match available space, in other scenarios the region gets sized via content size.
+
+- horizontalPosition - read only, the current horizontal position of the component. Possible values are 'left', 'right' or 'unset'.
+- verticalPosition - read only, the current vertical position of the component. Possible values are 'top', 'bottom' or 'unset'.
+
+*Slots:*
+- default slot for content
+
+*Events:*
+- positionchanged - event is thrown whenever the placement of the region relative to the anchor changes (top/bottom and left/right). 
+
+
+### Anatomy and Appearance
+**Structure:**
+```
+<div class="root">
```

Comment:
You probably don't need the "root" element - the root is just the `anchor-region` element

Reply:
One of the issues we had with the previous version was that resizing the region when it was rendered would cause the host page layout to jiggle so the universal solution was to wrap the region in a fixed size 0x0 div.  I'm proposing that we should do this by default which requires an fixed size wrapping div.